packages/flutter/lib/src/widgets/image.dart

```diff
@@ -305,6 +323,16 @@ class Image extends StatefulWidget {
   /// [FilterQuality.none] which corresponds to nearest-neighbor.
   ///
   /// If [excludeFromSemantics] is true, then [semanticLabel] will be ignored.
+  ///
+  /// If [cacheWidth] or [cacheHeight] are provided, it indicates to the
+  /// engine that the image should be decoded at the specified size. The image
+  /// will be rendered to the constraints of the layout or [width] and [height]
+  /// regardless of these parameters. These parameters are primarily intended
+  /// to reduce the memory usage of [ImageCache].
+  ///
+  /// In the case where the network image is on the Web platform, the [cacheWidth]
```

Comment:
See here for description on web behavior

Reply:
LGTM, but let's file an issue for us to implement it later. We can definitely implement it in the CanvasKit-based backend, and with some tricks in the HTML backend as well.